zerver/management/commands/deliver_scheduled_messages.py

```diff
@@ -62,12 +61,10 @@ def handle(self, *args: Any, **options: Any) -> None:
                 messages_to_deliver = ScheduledMessage.objects.filter(
                     scheduled_timestamp__lte=timezone_now(),
                     delivered=False)
-                if messages_to_deliver:
-                    for message in messages_to_deliver:
-                        with transaction.atomic():
-                            do_send_messages([self.construct_message(message)])
-                            message.delivered = True
-                    Message.objects.bulk_update(messages_to_deliver, ['delivered'])
```

Comment:
I think this was actually a bug here? This should have been `ScheduledMessage`, not `Message`, unless I'm missing something

Reply:
Good catch!  Updated the commit message to note this.

Reply:
@mateuszmandera would you be up for adding a test that would fail with the old code here?  I think basically we want to take this bit of code out of the management command and into a function we can call from tests, like you've done with other similar blocks of code.